politeiad/plugins/pi/pi.go

```diff
@@ -345,6 +348,115 @@ type SetBillingStatusReply struct {
 	Timestamp int64  `json:"timestamp"` // Unix timestamp
 }
 
+// Summary requests the summary of a proposal.
+type Summary struct {
+	Token string `json:"token"`
+}
+
+// SummaryReply is the reply to the Summary command.
+type SummaryReply struct {
+	Summary ProposalSummary `json:"summary"`
+}
+
+// ProposalSummary summarizes proposal information.
+//
+// StatusReason field will be populated if the status change required a
+// reason to be given. Examples include when a proposal is censored/abandoned
+// or when the billing status of the proposal is set to closed.
+type ProposalSummary struct {
+	Status       PropStatusT `json:"status"`
+	StatusReason string      `json:"statusreason"`
+}
+
+// PropStatusT represents the status of a proposal. It combines record and
+// plugin metadata in order to create a unified map of the various paths a
+// proposal can take throughout the proposal process. This serves as the
+// source of truth for clients so that they don't need to try and decipher
+// what various combinations of plugin metadata mean for the proposal.
+type PropStatusT string
+
+const (
+	// PropStatusInvalid represents an invalid proposal status.
+	PropStatusInvalid PropStatusT = ""
```

Comment:
Why isn't it "invalid" or whatnot?

This will print like crap.

Reply:
Updated 👍🏽